File: backend/Models/Salon.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace backend.Models
{
    public class Salon
    {
        public int Id { get; set; }
        [Required]
        public string Name { get; set; }

        [Required]
        public string Address { get; set; }

        [Required]
        public string City { get; set; }
        public int EmployeeNumber { get; set; }

        public byte[] Image { get; set; }
        public int Opened { get; set; }
        public int Closed { get; set; }

        public string PhoneNumber { get; set; }
        public string Email { get; set; }

        public List<User>? Users { get; set; }
        public ICollection<Appointment>? Appointments { get; set; }
        public ICollection<SalonTreatment> SalonTreatments { get; set; }
        public ICollection<Review>? SalonReviews { get; set; }



    }
}

File: frontend/src/app/models/salon.ts
import { Review } from "./review";
import { SalonTreatments } from "./salonTreatments";
import { User } from "./user";

export interface Salon {
  id: number;
  name: string;
  city: string;
  address: string;
  image?: string;
  employeeNumber: number;
  phoneNumber: string;
  email: string;
  opened: number;
  closed: number;
  employees?: User[];
  salonTreatments?: SalonTreatments[];
  salonReviews?: Review[];
  average?: number;
}
